src/index.js

```diff
@@ -1,89 +1,99 @@
+      const { githubUsername } = answers;
+      if (githubUsername === "") {
         console.log(
-          chalk.bgYellow.bold(
-            ` File with ${answers.name}.json already exists, try again! `
-          )
+          chalk.bgRed.bold(` Please enter a valid GitHub username. `)
         );
-        start();
-      } else {
-        checkUser(GHUsername).then((result) => {
-          if (result === true) {
-            questions(GHUsername);
-          } else {
+        init();
+      } else if (fs.existsSync(`./public/data/${githubUsername}.json`)) {
+        console.log(
+          chalk.bgYellow.bold(` File ${githubUsername}.json already exists!`)
+        );
+        prompt([
+          {
+            type: "confirm",
+            name: "overwrite",
+            message: "Do you want to overwrite the existing file?",
+          },
+        ]).then((answers) => {
+          const { overwrite } = answers;
+          if (overwrite) {
             console.log(
-              chalk.bgRed.bold(
-                ` User with username ${GHUsername} does not exist on GitHub, try again! `
-              )
+              chalk.bgGreen.bold(` Proceed with overwriting file... `)
             );
-            start();
+            start(githubUsername);
+          } else {
+            console.log(chalk.bgRed.bold(` File not overwritten! `));
+            console.log("Restart the program to try again.");
+            process.exit(0);
           }
         });
+      } else {
+        start(githubUsername);
       }
     })
-    .catch((error) => {
-      console.log(error);
-    });
-}
-
-function questions(username) {
-  prompt([
-    {
-      type: "input",
-      name: "name",
-      message: "What is your name?",
-    },
-    {
-      type: "select",
-      name: "type",
-      message: "Your profile type?",
-      choices: ["Personal", "Community"],
-    },
-    {
-      type: "input",
-      name: "bio",
-      message: "Add a short bio about yourself",
-    },
-  ])
-    .then((answers) => {
-      createUser(username, answers);
-    })
     .catch((err) => {
       console.log(err);
     });
 }
 
-async function checkUser(username) {
-  const axios = require("axios");
-  try {
-    const response = await axios.get(
-      `https://api.github.com/users/${username}`
-    );
-    if (response.status === 200) {
-      return true;
+let json;
+function start(githubUsername) {
+  checkUser(githubUsername).then((result) => {
+    if (result === true) {
+      questions().then((answers) => {
+        json = answers;
+        prompt([
+          {
+            type: "confirm",
+            name: "addLink",
+            message: "Do you want to add a link to your profile?",
+          },
+        ])
+          .then((addLink) => {
+            if (addLink) {
+              addlinks().then((links) => {
+                json.links = links;
+                createUser(githubUsername, json);
+              });
```

Comment:
To resolve the loop error maybe something like this can help but for me it throwing another error (https://bobbyhadz.com/blog/node-await-is-only-valid-in-async-function) when I try to implement into the main program
```js
const { prompt } = require("enquirer");

let links = [];

// My function
const myfunction = async function (bool) {
  while (bool) {
    let answers = await prompt([
      {
        type: "input",
        name: "url",
        message: "What is the URL of the link?",
      },
      {
        type: "confirm",
        name: "addLink",
        message: "Do you want to add another link?",
      },
    ]);
    links.push({
        url: answers.url,
      });
    if (answers.addLink) {
      
      myfunction();
    } else {
      return links;
    }
  }
};

// Start function
const start = async function (a, b) {
  const links = await myfunction(true);
  console.log(links);
};

// Call start
start();
```

```
$ npm start

> linkfree-cli@1.2.0 start
> node src/index.js

/workspaces/LinkFree-CLI/src/index.js:78
              json.links = await addlinks(true);
                           ^^^^^

SyntaxError: await is only valid in async functions and the top level bodies of modules
    at Object.compileFunction (node:vm:352:18)
    at wrapSafe (node:internal/modules/cjs/loader:1032:15)
    at Module._compile (node:internal/modules/cjs/loader:1067:27)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1157:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)
    at node:internal/main/run_main_module:17:47
```